newIDE/app/src/GameDashboard/Feedbacks/GameFeedback.js

```diff
@@ -0,0 +1,135 @@
+  const [showUnsolved, setShowUnsolved] = React.useState(false);
+  const [feedbacks, setFeedbacks] = React.useState<?Array<Comment>>(null);
+  const [isErrored, setIsErrored] = React.useState(false);
+
+  const displayedFeedbacks = feedbacks
+    ? showUnsolved
+      ? filterUnsolvedComments(feedbacks)
+      : feedbacks
+    : null;
+
+  const loadFeedbacks = React.useCallback(
+    async () => {
+      setIsErrored(false);
+      const { getAuthorizationHeader, profile } = authenticatedUser;
+      if (!profile) {
+        setIsErrored(true);
+        return;
+      }
+      try {
+        const feedbacks = await listComments(
+          getAuthorizationHeader,
+          profile.id,
+          {
+            gameId: game.id,
+            type: RETRIEVED_COMMENT_TYPE,
+          }
+        );
+        setFeedbacks(feedbacks);
+      } catch {
+        setIsErrored(true);
+      }
+    },
+    [authenticatedUser, game]
+  );
+
+  React.useEffect(
+    () => {
+      loadFeedbacks();
+    },
+    [loadFeedbacks]
+  );
+
+  return (
+    <Column noMargin expand>
+      <Line>
+        {!authenticatedUser.authenticated && (
+          <EmptyMessage>
+            <Trans>You need to login first to see your game feedbacks.</Trans>
+          </EmptyMessage>
+        )}
+        {authenticatedUser.authenticated &&
+          !displayedFeedbacks &&
+          !isErrored && <PlaceholderLoader />}
+        {authenticatedUser.authenticated && isErrored && (
+          <PlaceholderError onRetry={() => {}}>
+            <Text>
+              <Trans>
+                An error occured while retrieving feedbacks for this game.
+              </Trans>
+            </Text>
+          </PlaceholderError>
+        )}
+        {authenticatedUser.authenticated && displayedFeedbacks && (
+          <Column expand>
+            <Column>
+              <Line>
+                <Checkbox
+                  checked={showUnsolved}
+                  onCheck={() => {
+                    setShowUnsolved(!showUnsolved);
+                  }}
+                  label={<Trans>Show unsolved feedbacks only</Trans>}
+                />
+              </Line>
+            </Column>
+            {displayedFeedbacks.length === 0 && (
+              <EmptyMessage>
+                {showUnsolved ? (
+                  <Trans>
+                    You don't have any unsolved feedback for this game.
+                  </Trans>
+                ) : (
+                  <Trans>You don't have any feedback for this game.</Trans>
+                )}
+              </EmptyMessage>
+            )}
+            {displayedFeedbacks.length !== 0 && (
+              <ColumnStackLayout expand>
+                {displayedFeedbacks.map((comment: Comment) => (
+                  <FeedbackCard
+                    key={comment.id}
+                    comment={comment}
+                    authenticatedUser={authenticatedUser}
+                    onUpdateComment={loadFeedbacks}
```

Comment:
why do you need to load all feedbacks every time a comment is updated? This feels overkill / spamming the network.
Can we instead:
- always rely on the state of the component, which is initialised from the values in the first call to loadFeedbacks()
- when a check is done -> update the value in the state optimistically, whilst the network call happens
- if the network calls finished, fine, we do nothing
- if it fails, then the catch() reverts the value in the state with an error message

?

Reply:
Also I think the way it's done is already optimistic, from something I suggested, but Clement is surely right his approach is simpler/avoids reloading everything everytime (while I surely suggested the "overkill" solution which is to do the change in a state and then reload everything and then clear the state. While here a local change is good enough, no need to reload everything)